app/src/main/java/com/example/weather/DetailsFragment.kt

```diff
@@ -95,4 +82,23 @@ class DetailsFragment : Fragment() {
         val formatter = DateTimeFormatter.ofPattern("K:mm a", Locale.ENGLISH)
         return instant.atOffset(offset).format(formatter)
     }
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun updateBindings(binding: FragmentDetailsBinding, it: CityData?) {
+        binding.name.text = it?.name.toString()
+        binding.country.text = it?.sys?.country.toString()
+        binding.temp.text = convertCelsiusToFahrenheit(it?.main?.temp).toString() + "℉"
+        binding.lowAndHigh.text = formatLowAndHighDetails(it)
+        binding.humidity.text = it?.main?.humidity.toString() + "%"
+        binding.windspeedAmount.text = it?.wind?.speed.toString() + " m/s"
+        binding.pressureAmount.text = it?.main?.pressure.toString() + " hPa"
+        binding.description.text = it!!.weather[0].description[0].toUpperCase() + it!!.weather[0].description.substring(1)
+        binding.sunrise.text = convertTime(it?.sys?.sunrise, it?.timezone)
+        binding.sunset.text = convertTime(it?.sys?.sunset, it?.timezone)
+        Glide.with(this)
+            .load(getString(R.string.https_openweathermap_org_img_wn) + it!!.weather[0].icon + getString(
```

Comment:
for the future, strings stored in strings.xml are more used for UI text where as strings stored as CONSTs are more used in the code so you might wanna make these constants instead